src/main/java/org/sonarsource/scanner/cli/Main.java

```diff
@@ -71,18 +73,24 @@ void analyze() {
       checkSkip(p);
       configureLogging(p);
       init(p);
-      try (var engine = scannerEngineBootstrapper.bootstrap()) {
-        var success = engine.analyze((Map) p);
-        if (success) {
-          displayExecutionResult(stats, "SUCCESS");
-          status = Exit.SUCCESS;
+      try (var result = scannerEngineBootstrapper.bootstrap()) {
+        if (result.isSuccessful()) {
+          var engine = result.getEngineFacade();
+          var success = engine.analyze((Map) p);
+          if (success) {
+            displayExecutionResult(stats, SUCCESS);
+            status = Exit.SUCCESS;
+          } else {
+            displayExecutionResult(stats, FAILURE);
+            status = Exit.SCANNER_ENGINE_ERROR;
+          }
         } else {
-          displayExecutionResult(stats, "FAILURE");
-          status = Exit.SCANNER_ENGINE_ERROR;
+          displayExecutionResult(stats, FAILURE);
+          status = Exit.INTERNAL_ERROR;
```

Comment:
Looks like it's already done on line 70? (I have a Sonar issue in the IDE.)

Reply:
Sounds good.
I feel like the whole method could benefit from some refactoring, as it's starting to be a bit complex to read, IMO. But maybe not in this PR.